File: Examples/Example-Source/Table Cell Button/TableCellButtonExampleViewController.m
#import "TableCellButtonExampleViewController.h"
#import <Button/Button.h>

@interface TableCellButtonExampleViewController ()

@property (nonatomic, strong) BTNVenue *venue;
@property (nonatomic, strong) NSArray *items;

@end

@implementation TableCellButtonExampleViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.venue = [BTNVenue venueWithId:@"abc123"
                             venueName:@"Parm"
                              latitude:40.723027
                             longitude:-73.9956459];
    
    self.items = @[ @{ @"cell-identifier": @"your-cell", @"text": @"(212) 993-7189" },
                    @{ @"cell-identifier": @"your-cell", @"text": @"parmnyc.com" },
                    @{ @"cell-identifier": @"your-cell", @"text": @"View Menu" },
                    @{ @"cell-identifier": @"your-cell", @"text": @"Drive: 9 minutes" },
                    @{ @"cell-identifier": @"button-cell" },
                    @{ @"cell-identifier": @"your-cell", @"text": @"Mon-Wed, Sun", @"details": @"11:00 AM - 11:00 PM" },
                    @{ @"cell-identifier": @"your-cell", @"text": @"Thu-Sat", @"details": @"11:00 AM - Midnight" },
                    @{ @"cell-identifier": @"your-cell", @"text": @"Menus", @"details": @"Brunch, Lunch, Dinner, Desert" },
                    @{ @"cell-identifier": @"your-cell", @"text": @"Drinks", @"details": @"Beer, Wine, Full Bar, Cocktails" },
                    @{ @"cell-identifier": @"your-cell", @"text": @"Dining Options", @"details": @"Take-out, Delivery" },
                    @{ @"cell-identifier": @"your-cell", @"text": @"Reservations", @"details": @"NO" },
                    @{ @"cell-identifier": @"your-cell", @"text": @"Credit Cards", @"details": @"Yes (incl. Amex)" },
                    @{ @"cell-identifier": @"your-cell", @"text": @"Outdoor Seating", @"details": @"No" },];
    
}


#pragma mark - UITableViewDelegate

- (NSInteger)tableView:(UITableView *)tableView numberOfRowsInSection:(NSInteger)section {
    return self.items.count;
}


- (UITableViewCell *)tableView:(UITableView *)tableView cellForRowAtIndexPath:(NSIndexPath *)indexPath {
    
    NSDictionary *rowInfo = self.items[indexPath.row];
    UITableViewCell *cell = [tableView dequeueReusableCellWithIdentifier:rowInfo[@"cell-identifier"]
                                                            forIndexPath:indexPath];
    
    cell.textLabel.text       = rowInfo[@"text"];
    cell.detailTextLabel.text = rowInfo[@"details"];
    
    if ([cell isKindOfClass:[BTNDropinButtonCell class]]) {
        BTNDropinButtonCell *buttonCell = (BTNDropinButtonCell *)cell;
        [buttonCell prepareForDisplayWithVenue:self.venue completion:^(BOOL isDisplayable) {
            NSLog(@"Displayable: %@", @(isDisplayable));
            // You might remove the backing data item and reload your table view if the button cell is not displayable.
        }];
    }
    
    return cell;
}

@end

File: Examples/Swift/Example-Source/Table Cell Button/TableCellButtonExampleViewController.swift
import UIKit
import Button

class TableCellButtonExampleViewController: UITableViewController {

    let context: BTNContext = BTNContext.init(subjectLocation:
        BTNLocation.init(name: "Parm", latitude: 40.723027, longitude: -73.9956459));
    
    lazy var items: [[String:String]] = [
        ["cell-id": "your-cell",    "text": "(212) 993-7189"],
        ["cell-id": "your-cell",    "text": "parmnyc.com"],
        ["cell-id": "your-cell",    "text": "View Menu"],
        ["cell-id": "your-cell",    "text": "Drive: 9 minutes"],
        ["cell-id": "button-cell"],
        ["cell-id": "your-cell",    "text": "Mon-Wed, Sun",    "details": "11:00 AM - 11:00 PM"],
        ["cell-id": "your-cell",    "text": "Thu-Sat",         "details": "11:00 AM - Midnight"],
        ["cell-id": "your-cell",    "text": "Menus",           "details": "Brunch, Lunch, Dinner, Desert"],
        ["cell-id": "your-cell",    "text": "Drinks",          "details": "Beer, Wine, Full Bar, Cocktails"],
        ["cell-id": "your-cell",    "text": "Dining Options",  "details": "Take-out, Delivery"],
        ["cell-id": "your-cell",    "text": "Reservations",    "details": "NO"],
        ["cell-id": "your-cell",    "text": "Credit Cards",    "details": "Yes (incl. Amex)"],
        ["cell-id": "your-cell",    "text": "Outdoor Seating", "details": "No"]
    ]

    // MARK: - Table view data source

    override func tableView(_ tableView: UITableView, numberOfRowsInSection section: Int) -> Int {
        return items.count
    }

    override func tableView(_ tableView: UITableView, cellForRowAt indexPath: IndexPath) -> UITableViewCell {
        
        let rowInfo = items[(indexPath as NSIndexPath).row]

        let cell = tableView.dequeueReusableCell(withIdentifier: rowInfo["cell-id"]!, for: indexPath)
        
        cell.textLabel?.text = rowInfo["text"]
        cell.detailTextLabel?.text = rowInfo["details"]

        if let buttonCell = cell as? BTNDropinButtonCell {
            buttonCell.dropinButton?.borderWidth = 0.0
            
            // ERROR: Replace YOUR_BUTTON_ID with your Button ID from the Button Dashboard https://app.usebutton.com
            buttonCell.buttonId = "YOUR_BUTTON_ID"
            
            buttonCell.prepare(with: context, completion: { (isDisplayable: Bool) -> Void in
                print("Displayable: \(isDisplayable)")
            })
        }
        
        return cell
    }
    
}
